### ISSUE TITLE
Multiple document support

### ISSUE THREAD

**Issue by skaller** [rtfd/readthedocs.org#3204]
Provide a way to build multiple documents.

I am thinking higher level document index which specifies the individual documents.
Although this can be done with multiple repositories this prevents sharing
common files.

**Comment by agjohnson** [comment]
I'm not quite sure what pattern you're describing. It sounds like you're describing the basic functionality of Sphinx. Care to elaborate?

**Comment by skaller** [comment]
Sure, sorry if I was unclear. At present RTD presents a single document for my project Felix.
Say, Reference. But I want more documents, say, User Guide, Tutorial, Theory so now I need 4 separate documents.

These documents are not contained in each other. I can do this now with multiple
Git repositories, one for each document, but then I cannot share files between
them easily.

If I put the bodies of all these documents in one directory of one repository,
I can easily share files, but then I get a single top level table of contents
with a the expanding tab index on the left, spanning all the documents,
and I get a single document title.

The documents, however, are quite separate, handling different aspects
of a complex toolkit. I might want an FAQ for example, and the layout and
style would not be the same.

I would guess I need something like 20 documents. My system is a complete
programming language. So I need a language reference, a library reference,
tutorial guides, tool chain reference, installation guide, building guide,
contribution guide, embedding guide, compiler guide, advocacy documents,
collections of theoretical papers, and in some cases a complete book
on just one aspect of the system.

So roughly speaking I want to have multiple "conf.py" files each which builds
an independent document. Of course I want these documents to be able
to reference each other. So my thinking is a "docindex.py" file which itself
a master document that lists each of these documents.

Put another way, I seek a forest, not a tree. :)

**Comment by agjohnson** [comment]
No need for multiple conf.py files, this is all standard usage of Sphinx. I'd suggest looking more into suggested layout patterns for Sphinx, or looking at some other example projects using Sphinx.

If you end up really needing multiple projects, which it doesn't sound like you do, you can include multiple conf.py files in a single repository. Projects on RTD are specified with the hardcoded path to each conf.py.

Closing this as we can't help much here, and I think Sphinx already supports what you're looking to do with your documentation.

**Comment by skaller** [comment]
To be more specific, if you go here:

http://felix.readthedocs.io/en/latest/

you will see the start of a language reference using RTD.

You will see a table of contents on that page. That table of contents
is approximately 1% of the complete reference manual. It is already
too long. But that is just the reference manual for the language.

Now I want to do some tutorial chapters, tool guides, library documentation, etc:
all these documents already exist in another format. In fact many exist
in two formats which is a pain enough, and now I'm adding a third format.

It doesn't make sense for the table of contents to contain headings for
each of the above documents, each one should have its own table of
contents. In fact, many of these documents are already so large that
I may need to break them up. The reference manual in LaTeX format
is several hundred printed pages already, and a book on one specific
feature, coroutines, is also several hundred pages, documents only
three small library components, and a few test cases, and is likely
to run to thousands of pages if I don't factor it somehow .. and that's
a single topic.

**Comment by skaller** [comment]
So if I understand your comment correctly, I can have multiple RTD projects each using the same repository on Git, but linking to distinct conf.py files. If so that would solve my problem perfectly. But I don't know how to do that, the RTD docs just say RTD searches the repository for conf.py files, and indeed, there was another one present when I started and RTD kept building it instead of the one I wanted until I renamed it to force it to find the one I intended.

Since I asked this question on Stackoverflow it would help if specific instructions were given,
I would post your answer on Stackoverflow. It will save me a lot of hassle. I already have to deal with a hundreds of other toolchains, it really helps if an expert can just provide the details I need rather than having to investigate it myself.

**Comment by skaller** [comment]
OK, this issue was closed prematurely because the respondent failed to understand the issue, which was a RTD issue, not a Sphinx issue. However the mention of the fact that RTD hardlinks to the conf.py file, whilst incorrect, has lead to a solution. RTD searches for the conf.py file by default, however advanced settings allow it to be specified. 

I have now solved my problem exactly as I wish, by importing my repository multiple times as separate RTD projects, and setting the hard link to the distinct conf.py file in each case. (The only pain is that they have to be named conf.py which means they have to be in separate directories). 

This produces a set of completely independent documents, as required, whilst still allowing some files to be shared (via Sphinx includes) , and also allows a master index as yet another document, with ordinary HTML links as references to the separate RTD hosted documents. Multiple triggers are required on Github, one for each  RTD project, which means RTD, unfortunately, clones the repository multiple times. However it seems to rebuild every commit anyhow, even if the documents are not modified so I guess this isn't so bad.

[As an aside it may be worth adding a parse of commits for [rtd skip] in the message so the client can avoid triggering unnecessary rebuilds. Both Appveyor an Travis do this to avoid rebuilding projects when not required.]
